### PR TITLE
fully functional implementation of trakt Authentication 

### PR DESCRIPTION
**Pull request by geogolem**
and trakt private list support.

This code uses api.couchpota.to and poses as couchpotato (uses its
api_key) temporarily for testing purposes so that we can determine what
services radarr.aeonlucid.com will need to provide.

--the intention of this PR is not to have this code integrated right away. Work needs to be done in the backend @AeonLucid and I plan to figure that out soon.

I realize the dynamic object has to go.. and @onedr0p has made a few suggestions to improve the code especially w.r.t. the javascript - I will also look into this.

#### Database Migration
NO

#### Description
This implements trakt Authentication

#### Todos
- [ ] Tests
- [ ] Documentation

#### Issues Fixed or Closed by this PR

Trakt Authentication/Private Lists

### CODE REVIEW

**Comment by geogolem** [comment]
the previous PR related to this feature can be found here: https://github.com/Radarr/Radarr/pull/605 (just pasting for reference.

File: src/UI/Settings/NetImport/Options/NetImportOptionsView.js [hunk 1]
```diff
@@ -20,7 +16,11 @@ var view = Marionette.ItemView.extend({
 	},
 	
 	onShow : function() {
+		var params = new URLSearchParams(window.location.search.slice(1));
```
Comment (galli-leo) [inline src/UI/Settings/NetImport/Options/NetImportOptionsView.js]: Can you update that to not use URLSearchParams? Wont work in safari :(

Comment (geogolem) [inline src/UI/Settings/NetImport/Options/NetImportOptionsView.js]: image

ya sorry -- I always intended to swing back to this... just didnt have the chance yet -- will fix it now.

Comment (geogolem) [inline src/UI/Settings/NetImport/Options/NetImportOptionsView.js]: I committed a change that uses a polyfill: https://github.com/WebReflection/url-search-params as recomended by GOOGLE at the bottom of this page: https://developers.google.com/web/updates/2016/01/urlsearchparams?hl=en

I tested in IE.. It didnt work before the change, but works afterwards - could you please test in Safari...

Comment (geogolem) [inline src/UI/Settings/NetImport/Options/NetImportOptionsView.js]: image

I'm not sure if I need to guard var URLSearchParams = require('url-search-params');

with the a check like theone in the image.. and im not sure what this was referring to or if I need anything related to it:
image


I tested in Chrome/IE... and it seems to be working.

**Comment by geogolem** [comment]
there is a very, very rare bug... that could occur. The consequences of the bug are simply that your trakt authentication will fail until you re-authenticate.

This bug will only present itself if 3 things occur simultaneously:

1) A user happens to be on the netimport config settings page at the time his trakt tokens expires and
2) a netimport sync occurs (either automatically or because the user clicked the test button on a list settings  thing and
3) the user clicks save on the general settings

the problem here is that when the netimport sync occurs, the tokens get updated --> but these changes do not reflect in the GUI... if the user clicks save, his old tokens will be saved and they are no longer valid. so this bug can only happen if the user happens to be on the netconfig page when his tokens expire.

Im looking into a fix for this..
perhaps some kind of check before saving to prevent this.. or somehow getting the GUI to update the fields when the C# netimport sync refreshes the tokens... Im looking into this. I dont think its a big deal, because the workaround is to simply reauthenticate and it is most likely to be a rare occurance.. if you have any suggestions, please let me know.

**Pull request closed without merging**